controller/main.go

```diff
@@ -109,6 +131,226 @@ func (c Config) GetProvisionerPasswordPath() string {
 	return "/home/step/password/password"
 }
 
+// getSubject returns the identity being requested, so it can be validated
+func (req TLSIdentityRequest) getSubject() string {
+	return req.commonName
+}
+
+// mkBootstrapper generates a bootstrap container based on the template defined in Config. It
+// generates a new bootstrap token and mounts it, along with other required configuration, as
+// environment variables in the returned bootstrap container.
+func (req TLSIdentityRequest) mkBootstrapper(config *Config, fingerprint, podName, namespace string, provisioner *ca.Provisioner) (corev1.Container, error) {
+	b := config.Bootstrapper
+
+	token, err := provisioner.Token(req.commonName)
+	if err != nil {
+		return b, errors.Wrap(err, "tls token generation")
+	}
+
+	secretName, err := createTokenSecret(req.commonName+"-", namespace, token)
+	if err != nil {
+		return b, errors.Wrap(err, "create tls token secret")
+	}
+	log.Infof("TLS secret name is: %s", secretName)
+
+	b.Env = append(b.Env, corev1.EnvVar{
+		Name: "STEP_TOKEN",
+		ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: secretName,
+				},
+				Key: tokenSecretKey,
+			},
+		},
+	})
+	b.Env = append(b.Env, corev1.EnvVar{
+		Name:  "COMMON_NAME",
+		Value: req.commonName,
+	})
+	if req.duration != "" {
+		b.Env = append(b.Env, corev1.EnvVar{
+			Name:  "STEP_NOT_AFTER",
+			Value: req.duration,
+		})
```

Comment:
This will work but I still see references to the DURATION environment variable in bootstrapper/tls/bootstrapper.sh

With the environment variable named `STEP_NOT_AFTER` we don't need the if condition in the bootstrapper.sh